SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -27,58 +27,177 @@ This SWIP is part of a series of SWIPs (but can be implemented on its own). To s
 
 ## Specification
 <!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
 
-At a high level a payment module is responsible for:
+### Generic payment module
+At a high level a payment module is responsible for sending a payment to a recipient (a peer providing storage services) for services provided. How payments are actually performed will depend on the specific implementation of the payment module. The payment module is then responsible for:
 
-* Accepting an amount (in honey) and a recipient.
-* Resolving the conversion from honey to money (currently by querying the agreed-upon price oracle with the recipient.)
+* Resolving the conversion from honey to money.
 * Ensuring that the user can engage in SWAP accounting for the chosen payment module before payment is due.
-* Sending the recipient a payment.
-* Returning true when the payment was successful.
-* Returning false when the payment was not successful.
-* Verifying any on-chain conditions and making sure that they don't change within a reasonable time window after the transaction happens (due to block reorganizations / payment-module specific conditions)
-* Referencing a type, version and base currency 
+* Sending a payment to the recipient.
+* Sending a message to the accounting module upon receipt of payment
+* Ensuring that the user can send payments with the chosen payment module before payment is due.
+* Handling any potential errors.
 * Optionally exposing other methods such as querying balances, topping up balances or sending payments (outside of Swarm). 
 
-Nodes can specify their preference for both payment module, as well as price oracle in a list in a configuration file. The preferences are normalized and weighed.
-For any preference list, the chosen option will be the option which has the highest cumulative preference. The preference list has three dimensions, which will be resolved from high to low:
+### Allowing multiple payment modules
+When allowing multiple payment modules, it is essential for nodes to communicate their preference. There are three dimensions to take into consideration for the payment preference: 
 
 1. Currency to use
 2. Price oracle to use
-3. Payment method to use
+3. Payment provider to use
+
+We propose a standardized way to communicate these preferences and an algorithm to resolve any two preferences lists.
+
+For any dimension, nodes list the relative importance of them. Consequently, nodes specify their accepted options within each dimension. Importance is communicated through Swarm by any positive integer.
+
+The following is an example of the proposed configuration:
+
+```yaml
```

Comment:
Somehow this renders red markers on my pc. Is this also the case for you? 

![image](https://user-images.githubusercontent.com/33000441/63597840-a093a700-c5be-11e9-89a8-098d26ceaddd.png)

Reply:
Same for me, I don't know why. I will look into it to find a way to disable the red markers.

Reply:
I have the same problem in the oracle SWIPS. I think it means invalid syntax. Are you using TAB, or SPACE? Dived in the documentation of Github markdown and 4 spaces should create an indented code block.